packages/react-charting/src/components/DonutChart/Arc/Arc.tsx

```diff
@@ -107,6 +91,35 @@ export class Arc extends React.Component<IArcProps, IArcState> {
     const yValue = point.yAxisCalloutData || point.data || 0;
     return point.callOutAccessibilityData?.ariaLabel || (legend ? `${legend}, ` : '') + `${yValue}.`;
   };
+
+  private _renderArcValue = (className: string) => {
+    const { arc, data, innerRadius, outerRadius, showValuesInPercent, totalValue, hideValues } = this.props;
+
+    if (hideValues || Math.abs(data!.endAngle - data!.startAngle) < Math.PI / 12) {
+      return null;
+    }
+
+    const [base, perp] = arc.centroid(data);
+    const hyp = Math.sqrt(base * base + perp * perp);
+    const textRadius = Math.max(innerRadius!, outerRadius!) + 2;
```

Comment:
How can we add tests for this mathematical logic.

Reply:
put on hold until we find a good way to test these things